tc_aws/loaders/s3_loader.py

```diff
@@ -32,6 +34,11 @@ def load(context, url, callback):
         def handle_data(file_key):
             if not file_key or 'Error' in file_key or 'Body' not in file_key:
                 logger.warn("ERROR retrieving image from S3 {0}: {1}".format(key, str(file_key)))
+                # We will now handle the failure and return a 404.
+                result = LoaderResult()
```

Comment:
Not every failure should return a 404. I think we should test for the 404 from S3 before returning this.